### PR TITLE
 destroy function checks and support force-destroy

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
Signed-off-by: Prajyot Parab <prajot.parab@ibm.com>

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -590,10 +590,17 @@ function apply {
 # Destroy the cluster
 #-------------------------------------------------------------------------
 function destroy {
-  precheck
-  log "Running terraform destroy... please wait"
-  retry_terraform 2 destroy "$vars -input=false"
-  success "Done! destroy commmand completed"
+  if [ -d $ARTIFACTS_DIR ] && [ -f ./"$ARTIFACTS_DIR"/terraform.tfstate ]; then
+    if [[ $($TF state list -state=./"$ARTIFACTS_DIR"/terraform.tfstate | wc -l) -gt 0 ]]; then
+      question "Do you want to proceed with destroy?" "yes"
```
Comment (yussufsh) [inline deploy.sh:595]: Can you please add a `force-destroy` option. This will fail on CD pipelines.

**Review by yussufsh — CHANGES_REQUESTED** [review]

File: deploy.sh [hunk 2]
```diff
@@ -590,10 +591,18 @@ function apply {
 # Destroy the cluster
 #-------------------------------------------------------------------------
 function destroy {
-  precheck
-  log "Running terraform destroy... please wait"
-  retry_terraform 2 destroy "$vars -input=false"
-  success "Done! destroy commmand completed"
+  force=$1
```
Comment (yussufsh) [inline deploy.sh:594]: Why is this needed?

Comment (Prajyot-Parab) [reply deploy.sh]: Resolved

File: deploy.sh [hunk 3]
```diff
@@ -590,10 +591,17 @@ function apply {
...
 #-------------------------------------------------------------------------
 function destroy {
-  precheck
-  log "Running terraform destroy... please wait"
-  retry_terraform 2 destroy "$vars -input=false"
-  success "Done! destroy commmand completed"
+  if [ -d $ARTIFACTS_DIR ] && [ -f ./"$ARTIFACTS_DIR"/terraform.tfstate ]; then
+    if [[ $($TF state list -state=./"$ARTIFACTS_DIR"/terraform.tfstate | wc -l) -gt 0 ]]; then
+      [[ $FORCE_DESTROY == 0 ]] && question "Are you sure you want to proceed with destroy?" "yes"
+      if [[ "${value}" == "yes" || $FORCE_DESTROY == 1 ]]; then
```
Comment (yussufsh) [inline deploy.sh:597]: What is the value of `value` incase not force destroy? could be the answer of the last question.

**Review by yussufsh — APPROVED** [review]
